src/Features/VisualBasic/Portable/InlineParameterNameHints/InlineParameterNameHintsService.vb

```diff
@@ -0,0 +1,90 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Composition
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.Host.Mef
+Imports Microsoft.CodeAnalysis.InlineParameterNameHints
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.InlineParameterNameHints
+    <ExportLanguageService(GetType(IInlineParameterNameHintsService), LanguageNames.VisualBasic), [Shared]>
+    Friend Class InlineParameterNameHintsService
+        Inherits AbstractInlineParameterNameHintsService
+
+        <ImportingConstructor>
+        <Obsolete(MefConstruction.ImportingConstructorMessage, True)>
+        Public Sub New()
+        End Sub
+
+        Protected Overrides Function AddAllParameterNameHintLocations(semanticModel As SemanticModel, nodes As IEnumerable(Of SyntaxNode), spans As List(Of InlineParameterHint), cancellationToken As CancellationToken) As List(Of InlineParameterHint)
+            For Each node In nodes
+                cancellationToken.ThrowIfCancellationRequested()
+                Dim simpleArgument = TryCast(node, SimpleArgumentSyntax)
+                If Not simpleArgument Is Nothing Then
+                    If Not simpleArgument.IsNamed AndAlso simpleArgument.NameColonEquals Is Nothing AndAlso IsExpressionWithNoName(simpleArgument.Expression) Then
+                        Dim param = simpleArgument.DetermineParameter(semanticModel, allowParamArray:=False, cancellationToken)
+                        If param IsNot Nothing AndAlso param.Name.Length > 0 Then
+                            spans.Add(New InlineParameterHint(param.Name, simpleArgument.Span.Start))
+                        End If
```

Comment:
nit: you could share this code with c# by having a protected helper in teh base class.  up to you.